src/modules/worlds/sagas.ts

```diff
@@ -12,9 +19,31 @@ export function* worldsSaga() {
 }
 
 function* handlefetchWorldsWalletStatsRequest(action: FetchWalletWorldsStatsRequestAction) {
-  const { address } = action.payload
+  let address = action.payload.address
+
+  if (!address) {
+    address = yield select(getAddress)
+  }
 
   try {
+    if (!address) {
+      const {
+        success
+      }: {
+        success: ConnectWalletSuccessAction
+        failure: ConnectWalletFailureAction
+      } = yield race({
+        success: take(CONNECT_WALLET_SUCCESS),
+        failure: take(CONNECT_WALLET_FAILURE)
+      })
+
+      if (success) {
+        address = success.payload.wallet.address
+      } else {
+        throw new Error('An address is required')
+      }
+    }
```

Comment:
This seems to be assuming that if it doesn't have an address, the process to connect to it is happening while this saga is executing. Could it happen to this saga to be executed when there's no wallet and the process to connect to the wallet is not in progress?